File: 卡码/代码随想录/JS/单调栈/LeetCode_0739.js
//739. 每日温度
const dailyTemperatures = function(temperatures) {
	const res=new Array(temperatures.length).fill(0);
	const st=new Array(0);
	for(let i=0;i<temperatures.length;i++){
		while(st.length&&temperatures[i]>temperatures[st[st.length-1]]){
			res[st[st.length-1]]=i-st[st.length-1];
			st.pop();
		}
		st.push(i);
	}
	return res;
};

File: 卡码/代码随想录/1/10-单调栈/Java/src/com/kurumi/LeetCode_0739.java
package com.kurumi;

import java.util.Stack;

public class LeetCode_0739 {

}
class Solution {
  public int[] dailyTemperatures(int[] temperatures) {
  	int[] res=new int[temperatures.length];
  	Stack<Integer> st=new Stack<>();
  	for(int i=0;i<temperatures.length;i++) {
  		while(!st.empty()&&temperatures[i]>temperatures[st.peek()]) {
  			res[st.peek()]=i-st.peek();
  			st.pop();
  		}
  		st.push(i);
  	}
  	return res;
  }
}